### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by mdancho84** [business-science/pytimetk#2]
# Phase 1: MVP Package

Develop a minimal package with the most important functions. 

## Priority 1 - Import Data Frame Operations

- [ ] `summarise_by_time()` / `summarize_by_time()` - Probably will need `tk_index()` to extract time series index
- [ ] `future_frame()` - We will also need `tk_make_future_timeseries()` and `tk_make_timeseries()`
- [ ] `pad_by_time()`
- [ ] `slidify()`

## Priority 2 - Plot Time Series

- [ ] `plot_time_series()` - Not sure if we should go with `plotly` or `altair` for interactive mode. I feel we should go with `plotnine` for non-interactive. 

## Priority 3 - Augment Operations

Note - These functions should overwrite columns that are named the same in the input data frame. 

- [ ] `tk_augment_timeseries_signature()` - `tk_get_timeseries_signature()`
- [ ] `tk_augment_lags()` / `tk_agument_leads()` - Will need `lag_vec()` , `lead_vec()`
- [ ] `tk_augment_slidify()`

## Phase 2: Extend Sklearn

- [ ] Time Series Splitting / Cross Validation Functionality
- [ ] Preprocessors
- [ ] Vectorized Functions - Box Cox, 

## Phase 3: Additional Plotting Utilities

- [ ] Plot ACF Diagnostics
- [ ] Plot Anomalies Diagnostics
- [ ] Plot Seasonal Diagnostics
- [ ] Plot STL Diagnositics
- [ ] Plot Time Series Regression

## Phase 4: Fill in Function Gaps Where Needed

Add additional functionality that was not identified in Phases 1-3.

**Comment by mdancho84** [comment]
**Project Plan: Timetk Phase 1 (0.1.0):**  https://github.com/orgs/business-science/projects/1

Please let me know if you would like to contribute and I will set you up as an Outside Collaborator.

**Comment by isachng93** [comment]
Hi mdancho84,

Thanks for bringing timetk to python!

Would you consider to build-in <sktime> in phase 3, and X13ARIMA SEATS in phase 2?

**Comment by joshdunnlime** [comment]
A nice extra addition to the augment module would be: `tk.augment_periodic_spline`. This has can have accuracy benefits over Fourier encoding, but does come at the cost of many more features.

In the sklearn example below, we see 12 spline features vs 2 Fourier, but a significant increase in rmse and mae:
https://scikit-learn.org/stable/auto_examples/applications/plot_cyclical_feature_engineering.html#periodic-spline-features

**Comment by mdancho84** [comment]
@isachng93 

> Would you consider to build-in in phase 3, and X13ARIMA SEATS in phase 2?

We are evaluating modeling and forecasting next. Will keep you posted. It may be a separate package.
